train_gpt2.cu

```diff
@@ -2317,18 +2393,18 @@ void gpt2_multi_gpu_accumulate(GPT2* model, MultiGpuConfig* multi_gpu_config) {
 #endif
 }
 
-void gpt2_update(GPT2 *model, float learning_rate, float beta1, float beta2, float eps, float weight_decay, int t) {
+void gpt2_update(GPT2 *model, float learning_rate, float beta1, float beta2, float eps, float weight_decay, int t, size_t shard_num_parameters, size_t shard_offset) {
```

Comment:
Changed the `gpt2_update` to take in shard related variables. Also thought of moving the `MultiGpuConfig` inside the model. Since it introduces more changes didn't do it.